### PR TITLE
Blog - Using Bazel to Improve Your Automated Test Suite

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
[Link](https://642abe7136db807d99e41245--ephemeral-narwhal-3ac3b2.netlify.app/blog/bazel-and-automated-tests/)
@balapriyac

**Comment by msali123** [comment]
image
image

@jalletto there's an issue when I use "$0" it doesnt recognizes the code block due to error but as I use '$0' its working

The Preview link up is with  '$0'

File: blog/_posts/2023-05-17-bazel-and-automated-tests.md [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    assert expected_primes == actual_primes
+~~~
+
+Now you can run `pytest` and verify that all the test cases pass:
+
+~~~{.bash caption=">_"}
+$ pytest
+~~~
+
+~~~
```
Comment (balapriyac) [inline blog/_posts/2023-05-17-bazel-and-automated-tests.md:153]:
```suggestion
~~~{ caption = "Output"}
```

File: blog/_posts/2023-05-17-bazel-and-automated-tests.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+========================== 3 passed in 0.01s =====================
+~~~
+
+### Configuring Bazel
+
+To get started with Bazel in a project, you need to declare a workspace. To do so, create a file named `WORKSPACE` at the root of the project. Usually, an empty `WORKSPACE` file is enough for Bazel to recognize a workspace, but if you want to, you can have project-specific configurations in this file.
+
+For this particular project, you need to load the [Python rules](https://bazel.build/reference/be/python) by placing the following code in the `WORKSPACE` file:
+
+~~~{ caption="WORKSPACE.bazel"}
```
Comment (balapriyac) [inline blog/_posts/2023-05-17-bazel-and-automated-tests.md:172]:
```suggestion
~~~{ caption="WORKSPACE"}
```

Comment (balapriyac) [reply blog/_posts/2023-05-17-bazel-and-automated-tests.md:172]: `WORKSPACE` file is a text file containing the dependencies and the like.

File: blog/_posts/2023-05-17-bazel-and-automated-tests.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+INFO: Build completed successfully, 2 total actions
+//lib:test_prime                                  PASSED in 0.2s
+
+Executed 1 out of 1 test: 1 test passes.
+INFO: Build completed successfully, 2 total actions
+~~~
+
+As you can see, all the tests pass successfully.
+
+![success]({{site.images}}{{page.slug}}/success.png)\
```
Comment (balapriyac) [inline blog/_posts/2023-05-17-bazel-and-automated-tests.md:242]: Nice image!

File: blog/_posts/2023-05-17-bazel-and-automated-tests.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+INFO: Build completed, 1 test FAILED, 2 total actions
+~~~
+
+As you can see, it shows that the test fails. Bazel also creates a `bazel-testlogs` directory (among three other directories) where you can find more details about the tests that were run. The log will be stored in `bazel-testlogs/<target-name>/test.log`. In this case, it's `bazel-testlogs/lib/test_prime/test.log`:
+
+~~~{.bash caption=">_"}
+$ cat bazel-testlogs/lib/test_prime/test.log
+~~~
+
+~~~{ caption="Output}
```
Comment (balapriyac) [inline blog/_posts/2023-05-17-bazel-and-automated-tests.md:278]:
```suggestion
~~~{ caption="Output"}
```

**Review by balapriyac — CHANGES_REQUESTED** [review]
Thanks, Sameer. I've suggested a few changes.

**Comment by msali123** [comment]
@balapriyac 
[Changes done](https://642d58cd106c02196d3b07fe--ephemeral-narwhal-3ac3b2.netlify.app/blog/bazel-and-automated-tests/)

**Comment by balapriyac** [comment]
@jalletto Sameer faced a problem with " in the test run output, so he used ' instead. This is ready to merge otherwise.

**Review by jalletto — APPROVED** [review]

**Pull request merged**